ci/scenarios/vPool/add_remove_vpool_concurrently/main.py

```diff
@@ -73,36 +71,39 @@ def add_vpools(self):
                                                    'storage_ip': sr.ip,
                                                    'storagerouter_ip': sr.ip,
                                                    'vpool_name': vpool_name,
-                                                   'writecache_size': 10}})
+                                                   'writecache_size': 5}})
             tasks[vpool_name] = self.api.post(api='storagerouters/{0}/add_vpool'.format(sr.guid), data=data)
-
         for vpool_name, task in tasks.iteritems():
             self.LOGGER.info('-> {0} running: {1}'.format(vpool_name, self.api.wait_for_task(task)[0]))
             try:
                 self.vpools.append(VPoolHelper.get_vpool_by_name(vpool_name))
-            except RuntimeError as ex:
-                self.LOGGER.exception('Unable to find {0}: {1}'.format(vpool_name, ex))
+            except VPoolNotFoundError as ex:
+                self.LOGGER.exception('Unable to find {0}: {1}.'.format(vpool_name, ex))
 
         assert self.NUMBER_OF_VPOOLS == len(self.vpools), 'Failed to create {0} vpools: only {1} found!'.format(self.NUMBER_OF_VPOOLS, len(self.vpools))
 
     def remove_vpools(self):
         """
-        remove all vpools that have been previously added
+        Method to remove all vpools
         """
-        self.LOGGER.info("Starting to validate removal of vpools concurrently")
+        self.LOGGER.info("Starting to validate removal of vpools concurrently.")
         dels = {}
-        while len(self.vpools) > 0:
-            vpool = self.vpools.pop()
+        for vpool in self.vpools:
             dels[vpool.guid] = self.api.post(api='vpools/{0}/shrink_vpool'.format(vpool.guid), data=json.dumps({'storagerouter_guid': vpool.storagedrivers[0].storagerouter.guid}))
         for vpool_name, task in dels.iteritems():
             self.LOGGER.info('-> deleting vpool nr {0}: {1}'.format(vpool_name, self.api.wait_for_task(task)[0]))
```

Comment:
`wait_for_task` should be passed a timeout and check its result, whether it was successful or not